src/landpatterns/QFP.stanza

```diff
@@ -3,55 +3,190 @@ defpackage jsl/landpatterns/QFP:
+  span : Toleranced,
+  pitch : Double,
+  lead-length : Toleranced,
+  lead-width : Toleranced,
+  ) -> Quad-Lead-Profile:
+  val lead = QFP-Lead(lead-type = lead-type, length = lead-length, width = lead-width)
+  val profile = Lead-Profile(span = span, pitch = pitch, lead = lead)
+  #Quad-Lead-Profile(profile, None(), profile, None())
+
+public defstruct QFP <: Quad-Package :
+  doc: \<DOC>
+  Number of leads ignoring any depopulated leads and thermal leads
+  <DOC>
+  num-leads:Int with: (ensure => ensure-even-positive!, as-method => true)
+
+  doc: \<DOC>
+  Lead Profiles for all of the QFP's edges
+
+  By default the QFN uses the {@link SmallGullWingLeads} protrusion
+  type by default.
+  <DOC>
+  lead-profile:Quad-Lead-Profile with: (as-method => true)
+  doc: \<DOC>
+  Optional thermal lead for the package.
+  <DOC>
+  thermal-lead?:False|Shape with: (as-method => true)
+
+  doc: \<DOC>
+  Package Body for the Quad-based Package.
+  <DOC>
+  package-body:PackageBody with: (as-method => true)
+
+  doc: \<DOC>
+  Pad Planner for the Quad package
+
+  This provides a default pad planner that assumes that all lead
+  positions are active and provides a rectangle shaped pad for
+  all positions. The user can override this with their preferred
+  shape as desired.
+  <DOC>
+  pad-planner:PadPlanner with: (as-method => true)
+
+  doc: \<DOC>
+  Lead Numbering Scheme for the Quad Package
+
+  This provides a numbering scheme for the leads of the Quad
+  package. By default, it uses {@link Column-Major-Numbering}.
+  <DOC>
+  lead-numbering:Numbering with: (as-method => true)
+
+  doc: \<DOC>
+  Density Level for the Generated Package
+  <DOC>
+  density-level:DensityLevel with: (as-method => true)
+
+with:
+  printer => true
+  constructor => #QFP
+
+public defn QFP (
+  --
+  num-leads:Int,
+  lead-profile:Quad-Lead-Profile,
+  thermal-lead?:False|Shape,
+  package-body:PackageBody,
+  pad-planner:PadPlanner = RectanglePadPlanner,
+  lead-numbering:Numbering = select-quad-numbering(num-leads, lead-profile),
+  density-level:DensityLevel = DENSITY-LEVEL
   ):
-  name = to-string("QFP%_X%_" % [package-size, package-size])
-  make-qfp-landpattern(
-    num-pins, pitch,
-    lead-length, lead-width,
-    package-size,
-    density-level = density-level,
-    corner-pads? = corner-pads?,
-    thermal-pad? = thermal-pad?
-  )
\ No newline at end of file
+  #QFP(
+    num-leads, lead-profile, thermal-lead?,
+    package-body, pad-planner, lead-numbering,
+    density-level
+  )
+
+public defmethod name (pkg:QFP) -> String :
+  defn to-deci (v:Double) -> String:
+    val v* = to-int( v * 100.0 )
+    to-string("%_" % [v*])
+  val body = package-body(pkg)
+  val p = to-deci(pitch $ x-leads $ lead-profile(pkg))
+  val w = to-deci(typ $ width(body))
+  val l = to-deci(typ $ length(body))
+  val h = to-deci(typ $ height(body))
+  val n = num-leads(pkg) + (1 when thermal-lead?(pkg) is-not False else 0)
+  to-string("QFP%_P%_X%_X%_-%_N" % [p, w, l, h, n])
```

Comment:
IPC7351:

<img width="835" alt="Screenshot 2024-07-16 at 4 00 52 PM" src="https://github.com/user-attachments/assets/6894ced5-f5a6-46f3-943d-70e87ed9d64c">

I think the naming deviates - I can help get more explicit examples

Reply:
what does "lead span l1" and "lead span l2 nominal" mean here?

Reply:
The lead span is from the edge of the pin on one side to the edge of the pin on the other side. 

In the following image - there are unequal numbers of pins on either side:
![image](https://github.com/user-attachments/assets/3b6fad03-fd65-4a2d-8da1-e748844e123e)

Leadspan 1 is the longer leadspan (typically) and lead span 2 is the shorter one.

Reply:
The naming convention in the screenshot above differs from the datasheet, where they just use `LQFP100` in this case (100-pin, low-profile). Is there a general pattern or is this manufacturer-specific or something?

Reply:
The naming is often manufacturer specific and depends on process and features. Common names are:

1.  `LQFP` - low profile
2.  `TQFP` - thin
3.  `CQFP` - Ceramic
4.  `PQFP` - Plastic

Some of this is also a branding thing. Let's not worry to much about it. The user can override the `name` method if they want to change it to something else.

Reply:
Sounds good -- I just implemented "T" and "S" since those were in both sources you gave me.